--- openEuler-rpm-config/brp-ldconfig	2020-08-20 17:13:06.000000000 +0800
+++ openEuler-rpm-config/brp-ldconfig.new	2023-03-08 19:45:58.241036244 +0800
@@ -5,4 +5,12 @@
   exit 0
 fi
 
+riscv64_lp64d=0
+if [ "$HOSTTYPE" = "riscv64" -a -e $RPM_BUILD_ROOT/usr/lib64 -a ! -e $RPM_BUILD_ROOT/usr/lib64/lp64d ]; then
+    ln -s . $RPM_BUILD_ROOT/usr/lib64/lp64d
+    riscv64_lp64d=1
+fi
 /sbin/ldconfig -N -r "$RPM_BUILD_ROOT"
+if [ $riscv64_lp64d = 1 ]; then
+    rm -f $RPM_BUILD_ROOT/usr/lib64/lp64d
+fi
