From 749215e0573475d3f8085c5976ec0a82b703ccae Mon Sep 17 00:00:00 2001
From: Jingyun Hua <huajingyun@loongson.cn>
Date: Wed, 14 Dec 2022 07:00:44 +0000
Subject: [PATCH] fix config error for loongarch64

Signed-off-by: Jingyun Hua <huajingyun@loongson.cn>
---
 config.sub | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config.sub b/config.sub
index aca1d40..4d60f77 100644
--- a/config.sub
+++ b/config.sub
@@ -395,6 +395,7 @@ case $basic_machine in
 	| k1om-* \
 	| le32-* | le64-* \
 	| lm32-* \
+	| loongarch32-* | loongarch64-* | loongarchx32-* \
 	| m32c-* | m32r-* | m32rle-* \
 	| m68000-* | m680[012346]0-* | m68360-* | m683?2-* | m68k-* \
 	| m88110-* | m88k-* | maxq-* | mcore-* | metag-* \
-- 
2.33.0

