From 13fba7b02178c0196061965155c01061fce004fc Mon Sep 17 00:00:00 2001
From: orange-snn <songnannan2@huawei.com>
Date: Thu, 28 Jan 2021 16:35:20 +0800
Subject: [PATCH] fix brp ldconfig

---
 macros | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros b/macros
index 678136a..7ef0539 100644
--- a/macros
+++ b/macros
@@ -89,7 +89,7 @@
 
 # Build root policy macros. Standard naming:
 # convert all '-' in basename to '_', add two leading underscores.
-%__brp_ldconfig /usr/lib/rpm/%{_vendor}/brp-ldconfig
+%__brp_ldconfig /usr/lib/rpm/euleros/brp-ldconfig
 %__brp_compress /usr/lib/rpm/brp-compress
 %__brp_strip /usr/lib/rpm/brp-strip %{__strip}
 %__brp_strip_comment_note /usr/lib/rpm/brp-strip-comment-note %{__strip} %{__objdump}
-- 
2.27.0

